src/main.rs

```diff
@@ -102,16 +115,28 @@ async fn provision() -> Result<(), anyhow::Error> {
     )?;
     default_headers.insert(header::USER_AGENT, user_agent);
     let client = Client::builder()
-        .timeout(std::time::Duration::from_secs(30))
+        .timeout(std::time::Duration::from_secs(HTTP_TIMEOUT_SEC))
         .default_headers(default_headers)
         .build()?;
 
+    let imds_endpoint = azure_core::Url::parse(IMDS_ENDPOINT)?;
+    let imds_client = ImdsClient::builder(Arc::new(AzureCliCredential::new()))
+        .endpoint(imds_endpoint.clone())
+        .retry(RetryOptions::exponential(ExponentialRetryOptions {
```

Comment:
for communicating with IMDS during provisioning it's not recommended to use Exponential retry as it can severely affect provisioning latency.

Reply:
Done, thanks.